programmers/난이도별/level01.문자열_내_p와_y의_개수/HyeonJeong.java

```diff
@@ -0,0 +1,18 @@
+public class HyeonJeong {
+    public static boolean solution(String s) {
+        int count = 0;
+        char[] array = s.toUpperCase().toCharArray();
+        for (int i = 0; i < s.length(); i++) {
+            if (array[i] == 'P')
+                count += 1;
+            else if (array[i] == 'Y')
+                count -= 1;
+        }
+        return count == 0;

```

Comment:
```suggestion
        return s.chars()
                  .reduce(0, (count, c) -> {
                      if (c == 'P') return count + 1;
                      if (c == 'Y') return count - 1;
                      return count;
                  }) == 0;
```
reduce를 사용하면 이런식으로 표현할 수 있습니다!

Reply:
덕분에 reduce 사용법을 알게 되었습니다!!